### ISSUE TITLE
Allow tags starting with a digit

### ISSUE THREAD

**Issue by mactep** [rvben/rumdl#799]
### Problem or motivation

Based on :
- https://pymarkdown.readthedocs.io/en/stable/plugins/rule_md018/
- https://github.com/DavidAnson/markdownlint/blob/main/doc/md018.md
- https://obsidian.md/help/tags

There does not seem to be a rule against tags starting with digits. In fact, Obsidian docs state:

> Tags must contain at least one non-numerical character. For example, #1984 isn't a valid tag, but [#y1984](https://publish.obsidian.md/#y1984) is.

### Proposed solution

Should we allow tags to start with digits and follow the Obsidian docs? We could gate this with an option:

```toml
[MD018]
tags = true
start-with-digits = true # I don't know how to name things
```

### Alternatives considered

In my use case, my tag is `#3d_printing`, so I can't use `magiclink = true` as a workaround.

### Feature category

Improve existing rule

### Example markdown

```markdown
# Header

#3d_printing
```

### Contribution

- [ ] I would be willing to submit a PR to implement this feature

**Comment by rvben** [comment]
Done, and released in [v0.2.53](https://github.com/rvben/rumdl/releases/tag/v0.2.53) (`6237384f`). rumdl now follows the Obsidian rule you quoted, and it needed no new option: a tag is valid when it contains at least one non-numeric character, so only an all-digits tag is rejected.

Measured on the released binary:

| tag | MD018 |
| --- | --- |
| `#3d_printing` | not reported |
| `#1tag` | not reported |
| `#y1984` | not reported |
| `#2023-project` | not reported |
| `#1984` | reported |

You need one of two things for tags to be recognized at all, which is most likely why you hit this:

- `flavor = "obsidian"`, which recognizes tags with no further configuration, or
- `[MD018] tags = true`, if you want tag recognition under another flavor.

Without either, `#3d_printing` is still read as a heading missing its space.

On `start-with-digits`: I did not add it, because the Obsidian rule has nothing ambiguous to configure around, and because the all-digits case is load-bearing rather than incidental. With `tags = true` set, a genuine heading typo like `#1 Introduction` is still reported, precisely because `#1` is all digits and therefore not a tag. Accepting digit-only tags would take that with it. For comparison, `#Heading text` is already silent under `tags = true`, which is the cost you opt into when you turn tag recognition on.
